src/controller/tests/data_model/TestCommands.cpp

```diff
@@ -116,27 +115,32 @@ void DispatchSingleClusterCommand(const ConcreteCommandPath & aCommandPath, chip
         }
         else if (responseDirective == kSendMultipleSuccessStatusCodes)
         {
+            apCommandObj->AddStatus(aCommandPath, Protocols::InteractionModel::Status::Success,
+                                    "No error but testing status success case");
+
             // TODO: Right now all but the first AddStatus call fail, so this
             // test is not really testing what it should.
-            for (size_t i = 0; i < 4; ++i)
+            for (size_t i = 0; i < 3; ++i)
             {
-                apCommandObj->AddStatusAndLogIfFailure(aCommandPath, Protocols::InteractionModel::Status::Success,
-                                                       "No error but testing AddStatusAndLogIfFailure in success case");
+                (void) apCommandObj->FailableAddStatus(aCommandPath, Protocols::InteractionModel::Status::Success,
+                                                       "No error but testing status success case");
             }
             // And one failure on the end.
-            apCommandObj->AddStatus(aCommandPath, Protocols::InteractionModel::Status::Failure);
+            (void) apCommandObj->FailableAddStatus(aCommandPath, Protocols::InteractionModel::Status::Failure);
         }
         else if (responseDirective == kSendError)
         {
             apCommandObj->AddStatus(aCommandPath, Protocols::InteractionModel::Status::Failure);
         }
         else if (responseDirective == kSendMultipleErrors)
         {
+            apCommandObj->AddStatus(aCommandPath, Protocols::InteractionModel::Status::Failure);
```

Comment:
This is going to make the test fail, since right now it expects a success status because of the todo.

But more generally: I wonder whether we need a public FallibleAddStatus at all, or whether we should just fix this test to test what it's _actually_ testing and leave the todo comments about how it should test something else.  And if FallibleAddStatus is internal-only, it could return Status, since that's what all its consumers want, and we could save a bit of codesize....

Reply:
I believe this does no changes from the previous state: we used to have 4 failures and a success.
I kept the same thing, but kept the first failure state as unconditional (because it does not fail) and set all the other ones to accept errors (since this is what they did).

I don't think this test actually validates what it thinks it does (we even have a comment about that) however that is not something that I changed.

Reply:
This test definitely does not validate what it thinks it does, we agree on that.

I see now, you moved the first loop iteration out.  OK, that makes sense, and I was just misreading things....